web3pi_proxy/core/rpc/node/endpoint_pool/pool_manager.py

```diff
@@ -92,14 +90,114 @@ def check_connections(self, pools: List[EndpointConnectionPool]):
                     daemon=True,
                 ).start()
 
-            if self.__is_outdated(pool):
-                self.__logger.warning(f"Endpoint `{pool.endpoint.name}`: falling behind")
-
             self.__logger.debug(f"Endpoint `{pool.endpoint.name}`: okay.")
 
 
+class HttpResponseParserListenerSyncController:
+    body: bytes = b''
+    block_number: int = 0
+    block_timestamp: int = 0
+    __logger = get_logger("SyncController")
+
+    def __init__(self, __pool_name: str):
+        super().__init__()
+        self.__pool_name = __pool_name
+
+    def on_body(self, body: bytes):
+        self.body = self.body + body
+
+    def on_message_complete(self):
+        try:
+            block_data = json.loads(self.body)
+            result = block_data.get('result')
+            if not result:
```

Comment:
I'd raise exceptions here and catch them in one place (in the SyncController) -> then, there would be a lot less if/then/etc logic responsible mostly for logging of different responses... :/